### DISCUSSION TITLE
which dir to put lsp configs

### DISCUSSION THREAD

**Discussion by bryant-the-coder** [williamboman/nvim-lsp-installer#669] [Q&A] [answered]
As show in @williamboman config, he place his lsp configurations in https://github.com/williamboman/nvim-config/tree/main/after/plugin/lsp/servers

So which is the best way and why?

**Comment by williamboman** [comment, marked as answer]
Hello! It really doesn't matter, this is entirely up to your preferences. I put mine under the `plugin/` (recently I moved this into `after/plugin/`). These files are sourced by neovim automatically according to `:h runtimepath`

**Reply by bryant-the-coder** [reply]
Thanks. But why put it in `after/plugin/` any advantages?

**Reply by williamboman** [reply]
The reason was that I wanted to try out decoupling plugin setup from the plugin manager (i.e. not depend on `config`, `setup` hooks that packer provides). The reason it's in the `after` directory is that I want to make sure it executes as late as possible during initialization, and the reason I want to run it as late as possible is to effectively make sure any code executing in `after/` has precedence (because it executed last).
